src/database/DataTypes/Transaction.js

```diff
@@ -142,13 +150,58 @@ export class Transaction extends Realm.Object {
    */
   pruneRedundantItems(database) {
     const itemsToPrune = [];
-    this.items.forEach((transactionItem) => {
-      if (transactionItem.totalQuantity === 0) itemsToPrune.push(transactionItem);
+    const transactionBatchesToPrune = [];
+    this.items.forEach(transactionItem => {
+      if (transactionItem.totalQuantity === 0) {
+        itemsToPrune.push(transactionItem);
+        transactionBatchesToPrune.concat(transactionItem.batches);
+      }
     });
+    database.delete('TransactionBatch', transactionBatchesToPrune);
```

Comment:
Actually this is unnecessary - the ```TransactionItem``` takes care of deleting any batches attached to it